osu.Game/Screens/Import/ReplayMissingBeatmapScreen.cs

```diff
@@ -0,0 +1,225 @@
+                            Anchor = Anchor.Centre,
+                            Origin = Anchor.Centre,
+                            Spacing = new Vector2(15),
+                            Children = new Drawable[]
+                            {
+                                new OsuSpriteText
+                                {
+                                    Text = "Beatmap info",
+                                    Font = OsuFont.Default.With(size: 30),
+                                    Anchor = Anchor.Centre,
+                                    Origin = Anchor.Centre,
+                                },
+                                new FillFlowContainer
+                                {
+                                    AutoSizeAxes = Axes.Both,
+                                    Direction = FillDirection.Horizontal,
+                                    Anchor = Anchor.Centre,
+                                    Origin = Anchor.Centre,
+                                    Spacing = new Vector2(15),
+                                    Children = new Drawable[]
+                                    {
+                                        beatmapPanelContainer = new Container
+                                        {
+                                            AutoSizeAxes = Axes.Both,
+                                            Anchor = Anchor.CentreLeft,
+                                            Origin = Anchor.CentreLeft,
+                                        },
+                                    }
+                                },
+                                automaticDownload = new SettingsCheckbox
+                                {
+                                    LabelText = "Automatically download beatmaps",
+                                    Current = config.GetBindable<bool>(OsuSetting.AutomaticallyDownloadWhenSpectating),
+                                    Anchor = Anchor.Centre,
+                                    Origin = Anchor.Centre,
+                                },
+                                replayDownloadButton = new ReplayDownloadButton(new ScoreInfo())
+                                {
+                                    Anchor = Anchor.Centre,
+                                    Origin = Anchor.Centre,
+                                },
+                            }
+                        }
+                    }
+                },
+            };
+        }
+
+        protected override void LoadComplete()
+        {
+            base.LoadComplete();
+
+            if (beatmapSetInfo == null)
+            {
+                var onlineBeatmapRequest = new GetBeatmapSetRequest(beatmap.OnlineBeatmapSetID);
+
+                onlineBeatmapRequest.Success += res =>
+                {
+                    beatmapSetInfo = res;
+                    updateStatus();
+                };
+                api.Queue(onlineBeatmapRequest);
```

Comment:
Is this _second_ online request necessary? When is this ever gonna be `null`?

Reply:
`BeatmapSet` in `APIBeatmap` is nullable?
I don't know if it will be null so I keep this logic

Reply:
It is nullable only because `APIBeatmap` is a structure reused by many endpoints, some of which will not return the beatmap set data by default. The lookup endpoint will, though, so this null path is essentially dead code.

I think it would be best to remove it and reduce some of the complexity in here.